articles/css-flexbox-vs-css-grid/index.md

```diff
@@ -0,0 +1,748 @@
+
+```html
+<div>
+    <div>One</div>
+    <div>Two Two</div>
+    <div>Three Three Three</div>
+    <div>Four Four Four Four</div>
+    <div>Five Five Five Five Five</div>
+</div>
+```
+
+In this example, the `div` is a container element where other `divs` inside the main `div` are the container items.
+
+This guide will discuss the differences and relationships that exist between Grid and Flexbox layouts.
+
+To get along with this guide, prior knowledge of [CSS Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) and [CSS Grid](https://css-tricks.com/snippets/css/complete-guide-grid/) will be of great importance.
+
+### What is CSS FlexBox
+It is a layout mode in CSS3. It gives a more proficient approach to designing, aligning, and disseminating space between items in a container to control their arrangement.
+
+With Flexbox, we can arrange items left to right, right to left, top to bottom, or bottom to top and at the same time control the spacing, alignment, and order of items in the container.
+
+### What is CSS  Grid
+CSS Grid is a collection of styles that allow you to control page layout based on rows and columns. In a Grid system, the containing elements are used to define a row within the layout. Then you apply class attributes to item elements that then span them over the desired number of columns. This allows you to create more complex nested layouts.
+
+### CSS Grid vs CSS FlexBox
+
+Let's have a look at some of the common differences between the two layouts.
+
+#### One vs two dimensions
+
+Flexbox arranges items in a one-dimensional layout. This can be either in rows or columns.
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-horizontal-or-vertical.png)
+
+A Flexbox layout fits best when aligning items either horizontally or vertically. It is one dimensional and can only fit a row or a column.
+
+![](/engineering-education/css-flexbox-vs-css-grid/vertical.png)
+
+![](/engineering-education/css-flexbox-vs-css-grid/horizontal.png)
+
+A perfect example of one dimensional is a header navigation bar or a footer.
+
+Let's have a look at a simple example of implementing a header navigation menu.
+
+```html
+<div class="nav">
+    <div>HOME</div>
+    <div>ABOUT</div>
+    <div>CONTACT</div>
+    <div>SIGN UP</div>
+    <div>SIGN IN</div>
+</div>
+```
+
+Some CSS to style the header.
+
+```css
+.nav {
+    display: flex;
+    background: seagreen;
+    height: 50px;
+    align-items: center;
+    gap: 10px;
+}
+
+.nav div {
+    color: whitesmoke;
+    margin-left: 20px;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-hearder-nav.png)
+
+Having a header in your pages, you would probably lay its items in one direction. It is flexibly easier to implement this with a Flexbox layout as it arranges items in one direction. The menu is one-dimensional that perfectly fits well with `display: flex`.
```

Comment:
```suggestion
Most common header layouts lay its items in one direction thus it's much easier to use Flexbox because it arranges items in one direction.
```